doc/contributing/documentation_guide.md

```diff
@@ -217,57 +217,83 @@ For methods written in C, \RDoc cannot determine what arguments
 the method accepts, so those need to be documented using \RDoc directive
 [`:call-seq:`](rdoc-ref:RDoc::Markup@Method+arguments).
 
+For a singleton method, use the form:
+
+```c
+class_name.method_name(method_args) {|block_args| ... } -> return_type
+```
+
 Example:
 
 ```c
-*  call-seq:
-*    array.count -> integer
-*    array.count(obj) -> integer
-*    array.count {|element| ... } -> integer
+*  :call-seq:
+*    Hash.new(default_value = nil) → new_hash
+*    Hash.new {|hash, key| ... } → new_hash
 ```
 
-When creating the `call-seq`, use the form
+For an instance method, use the form
+(omitting any prefix, just as RDoc does for a Ruby-coded method):
+
+```
+method_name(method_args) {|block_args| ... } -> return_type
+```
+Examples:
 
+```c
+*  :call-seq:
+*    count -> integer
+*    count(obj) -> integer
+*    count {|element| ... } -> integer
 ```
-receiver_type.method_name(arguments) {|block_arguments|} -> return_type
+
+```c
+* :call-seq:
+*    <=> other -< -1, 0, 1, or nil
 ```
 
-Omit the parentheses for cases where the method does not accept arguments,
-and omit the block for cases where a block is not accepted.
+Arguments:
 
-In the cases where method can return multiple different types, separate the
-types with "or".  If the method can return any type, use "object".  If the
-method returns the receiver, use "self".
+- If the method does not accept arguments, omit the parentheses.
+- If the method accepts optional arguments:
 
-In cases where the method accepts optional arguments, use a `call-seq` with
-an optional argument if the method has the same behavior when an argument is
-omitted as when the argument is passed with the default value.  For example,
-use:
+    - Separate each argument name and its default value with ` = `
+      (equal-sign with surrounding spaces).
+    - If the method has the same behavior with either an omitted
+      or an explicit argument, use a `call-seq` with optional arguments.
+      For example, use:
 
-```
-obj.respond_to?(symbol, include_all=false) -> true or false
-```
+        ```
+        respond_to?(symbol, include_all = false) -> true or false
+        ```
 
-Instead of:
+    - If the behavior is different with an omitted or an explicit argument,
+      use a `call-seq` with separate lines.
+      For example, use:
 
-```
-obj.respond_to?(symbol) -> true or false
-obj.respond_to?(symbol, include_all) -> true or false
-```
+        ```
+        default → object
+        default(key) → object
```

Comment:
The description says:

> If the behavior is different with an omitted or an explicit argument

But it looks like the behavior is the same regardless of whether the argument is present or not?

Reply:
Poorly chosen example.  What's a better one?

Reply:
For example, `Enumerable#max` is:

```
 *    max     -> element
 *    max(n)  -> array
```

Reply:
Got it, thanks.